kedro_partitioned/pipeline/decorators/decorators.py

```diff
@@ -12,6 +12,85 @@
 from kedro_partitioned.utils.typing import IsFunction
 from kedro_partitioned.utils.other import kwargs_only, identity
 from kedro_partitioned.utils.iterable import tolist
+from kedro_partitioned.utils.string import (
+    get_filepath_without_extension,
+)
+
+
+def multithread_partitions(
+    partitioned_args: List[str],
+):
```

Comment:
```suggestion
) -> t.Callable[[FN], FN]:
```

In the utils.typing file, define a typevar like this: `t.TypeVar("FN", bound=t.Callable[[...], t.Any])`